src/main/java/pl/umk/mat/git2befit/service/workout/factory/implementation/SplitTrainingPlan.java

```diff
@@ -88,65 +93,138 @@ private Training getTrainingForDay(Map<String, List<ExerciseExecution>> training
     }
 
     private int getAmountOfExercisesForBodyPart(String bodyPart) {
-        int amountOfExercises;
         if (smallBodyParts.contains(bodyPart))
-            amountOfExercises = 3;
+            return 3;
         else
-            amountOfExercises = 4;
-        return amountOfExercises;
+            return 4;
     }
 
-    private ExerciseExecution getUniqueExercise(List<Exercise> exercisesWithEquipmentFiltered, List<Exercise> exercisesWithoutEquipmentFiltered) throws IllegalStateException {
-        ExerciseExecution exerciseExecution = new ExerciseExecution();
-        Random random = new Random();
-        int randomInt;
-
-        if (isEnoughExercises(exercisesWithEquipmentFiltered)) {
-            randomInt = random.nextInt(exercisesWithEquipmentFiltered.size());
-            exerciseExecution.setExercise(exercisesWithEquipmentFiltered.get(randomInt));
-            exercisesWithEquipmentFiltered.remove(randomInt);
-        } else if (isEnoughExercises(exercisesWithoutEquipmentFiltered)) {
-            randomInt = random.nextInt(exercisesWithoutEquipmentFiltered.size());
-            exerciseExecution.setExercise(exercisesWithoutEquipmentFiltered.get(randomInt));
-            exercisesWithoutEquipmentFiltered.remove(randomInt);
-        }else {
+    private ExerciseExecution getUniqueExercise(List<Exercise> exercisesWithEquipmentFilteredByBodyPart,
+                                                List<Exercise> exercisesWithoutEquipmentFilteredByBodyPart) throws IllegalStateException {
+        var exerciseExecution = new ExerciseExecution();
+        var random = new Random();
+        int randomExerciseIndex;
+
+        if (isEnough(exercisesWithEquipmentFilteredByBodyPart)) {
+            randomExerciseIndex = random.nextInt(exercisesWithEquipmentFilteredByBodyPart.size());
+            exerciseExecution.setExercise(exercisesWithEquipmentFilteredByBodyPart.remove(randomExerciseIndex));
+        } else if (isEnough(exercisesWithoutEquipmentFilteredByBodyPart)) {
+            randomExerciseIndex = random.nextInt(exercisesWithoutEquipmentFilteredByBodyPart.size());
+            exerciseExecution.setExercise(exercisesWithoutEquipmentFilteredByBodyPart.remove(randomExerciseIndex));
+        } else {
             throw new IllegalStateException();
         }
 
+        return addSeriesAndCount(exerciseExecution);
+    }
+
+    private boolean isEnough(List<Exercise> exercises) {
+        return exercises.size() != 0;
+    }
+
+    private ExerciseExecution addSeriesAndCount(ExerciseExecution exerciseExecution) {
         exerciseExecution.setSeries(3);
         exerciseExecution.setCount(8);
-
         return exerciseExecution;
```

Comment:
Ta metoda w sumie przyjmuje ExerciseExecution, ustawia w nim 2 rzeczy i zwraca to samo. Czy nie lepiej żeby przekazać tu samo ćwiczenie, zrobić nowy obiekt ExerciseExecution, ustawić wszystko i zwrócić?